public/src/components/channelManagement/richTextEditor.tsx

```diff
@@ -0,0 +1,407 @@
+import React, { useCallback, useEffect, useLayoutEffect, useMemo, useState } from 'react';
+import {
+  BoldExtension,
+  ItalicExtension,
+  LinkExtension,
+  ShortcutHandlerProps,
+  TextHighlightExtension,
+  createMarkPositioner,
+} from 'remirror/extensions';
+import {
+  ComponentItem,
+  EditorComponent,
+  FloatingToolbar,
+  FloatingWrapper,
+  Remirror,
+  ToolbarItemUnion,
+  useActive,
+  useAttrs,
+  useChainedCommands,
+  useCurrentSelection,
+  useExtension,
+  useRemirror,
+  useUpdateReason,
+} from '@remirror/react';
+
+import { createStyles, WithStyles, withStyles } from '@material-ui/core';
+
+// CSS
+// Note that ReMirror/ProseMirror CSS has been added to the /src/style.css file
+const styles = createStyles({
+  fieldLabel: {
+    display: 'inline-block',
+    fontSize: '85%',
+    color: 'rgba(0 0 0 / 0.6)',
+    margin: '0 1.5em',
+  },
+  helperText: {
+    fontSize: '85%',
+    color: 'rgba(0 0 0 / 0.6)',
+    margin: '0.5em 0 0 1.5em',
+  },
+  errorText: {
+    color: 'rgba(0 0 0 / 1)',
+    backgroundColor: 'rgba(255 255 0 / 1)',
+    margin: '0.5em 0 0 1.5em',
+  },
+});
+
+// Typescript
+interface RichTextEditorProps extends WithStyles<typeof styles> {
+  disabled: boolean;
+  label?: string;
+  helperText?: string;
+  name?: string;
+  error: boolean;
+  updateCopy: (item?: string[]) => void;
```

Comment:
It would be nice if the types of the input value and the update value were consistent, and also if we could be explicit about whether we're working with `string` or `string[]`. This would avoid the need for e.g. `highlightedText: updatedHighlightedText[0]` throughout the codebase.

What about if the `RichTextEditor` deals only with `string`, which may contain html such as `<p>`.
And then for the special case of the `paragraphs` fields there's a wrapper component (`ParagraphsRichTextEditor`?) which deals with `string[]` and just does the transformations for the `<p>` tags, and wraps `RichTextEditor`.

Reply:
Just to illustrate what I mean -
```
interface ParagraphsRichTextEditor {
  paragraphs: string[];
  onUpdate: (value: string[]) => void;
  ...
}
const ParagraphsRichTextEditor: React.FC<ParagraphsRichTextEditor> = ({
  paragraphs,
  onUpdate,
  ...
}) => {
  return (<RichTextEditor
    value={parseCopyForParagraphs(paragraphs)}
    onUpdate={(updatedValue: string) => onUpdate(htmlToParagraphs(updatedValue))}
    ...
  />)
}
```

Reply:
Given that I've already done work to update banners to accept a replacement `paragraphs` attribute to (eventually) replace `messageText` in the SDC repo, I've gone with a solution that gets rid of code like `highlightedText: updatedHighlightedText[0]` to `highlightedText: updatedHighlightedText.join(' ')`.  It could be refactored, but I'd rather get this solution working first.

Reply:
I don't think the work to add `paragraphs` to banners is very relevant to this suggestion.
The various uses of the RTE will be simpler, and the types clearer, if `RichTextEditor` deals only with `string`. A simple wrapper component that does the transformation for <p> tags can be used where a `string[]` is required.
I'm happy to pair on this.